src/rs.cpp

```diff
@@ -308,6 +308,13 @@ void rs2_delete_device(rs2_device* device) BEGIN_API_CALL
 }
 NOEXCEPT_RETURN(, device)
 
+int rs2_device_is_connected( const rs2_device * device, rs2_error ** error ) BEGIN_API_CALL
+{
+    VALIDATE_NOT_NULL( device );
+    return device->device->is_valid();
```

Comment:
Check the internal ptr as well
`VALIDATE_NOT_NULL(device->device);`

Reply:
That'll be a crash, too -- if `device` is not null, then `device->device` is not null.
But I'll add.